src/web/components/ClientSideTokenGeneration/CstgAddDomainDialog.tsx

```diff
@@ -0,0 +1,86 @@
+import { useEffect, useState } from 'react';
+import { FormProvider, useForm } from 'react-hook-form';
+import { parse } from 'tldts';
+
+import { AddDomainNamesFormProps } from '../../services/domainNamesService';
+import { extractTopLevelDomain, separateStringsCommaSeparatedList } from '../../utils/textHelpers';
+import { Dialog } from '../Core/Dialog';
+import { InlineMessage } from '../Core/InlineMessage';
+import { TextInput } from '../Input/TextInput';
+
+import '../KeyPairs/KeyPairDialog.scss';
+
+type AddDomainNamesDialogProps = Readonly<{
+  onAddDomains: (newDomainNamesFormatted: string[]) => Promise<void>;
+  onOpenChange: () => void;
+}>;
+
+type DomainProps = {
+  isIcann: boolean | null;
+  isPrivate: boolean | null;
+  domain: string | null;
+};
+
+type ValidDomainProps = Omit<DomainProps, 'domain'> & { domain: string };
+
+function CstgAddDomainDialog({ onAddDomains, onOpenChange }: AddDomainNamesDialogProps) {
+  const [open, setOpen] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<string>();
+
+  const formMethods = useForm<AddDomainNamesFormProps>();
+  const { handleSubmit, setValue, reset } = formMethods;
+
+  useEffect(() => {
+    if (!open) {
+      setValue('newDomainNames', '');
+      setErrorMessage('');
+      reset();
+    }
+  }, [open, setValue, reset]);
+
+  const isValidDomain = (domainProps: DomainProps): domainProps is ValidDomainProps => {
```

Comment:
This can probably be moved to textHelpers.ts.  It will probably be used in the edit dialog, too